Build a __torch_dispatch__ class that records torch operator names

ttanpcs opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #78835
- Created a SchemaCheckTensor torch.tensor subclass to verify that ops match corresponding schema. Right now this keeps a running list of ops which can be cleared and displayed with start_recording() and display_ops() respectively within the schema_check_tensor class
- Created a test_schema_check test suite for testing SchemaCheckTensor and related functionality
- Specifically tested that SchemaCheckTensor acts/operates like a normal tensor and that SchemaCheckTensor writes ops in expected order.

Commit:
Build a __torch_dispatch__ class that records torch operator names

[ghstack-poisoned]

Commit:
Update on "Build a __torch_dispatch__ class that records torch operator names"

- Created a SchemaCheckTensor torch.tensor subclass to verify that ops match corresponding schema. Right now this keeps a running list of ops which can be cleared and displayed with start_recording() and display_ops() respectively within the schema_check_tensor class
- Created a test_schema_check test suite for testing SchemaCheckTensor and related functionality
- Specifically tested that SchemaCheckTensor acts/operates like a normal tensor and that SchemaCheckTensor writes ops in expected order. 

[ghstack-poisoned]

Gamrix commented on torch/testing/_internal/schema_check_tensor.py:
```diff
@@ -0,0 +1,52 @@
+
+ops = []
+
+class SchemaCheckTensor(torch.Tensor):
```
Could you add some brief documentation about what this class does and how it is to be used?

Gamrix reviewed (approved):
Overall looks good. Added a few comments in the diff about various nits that should be resolved.
Also, don't forget to fix the lintrunner errors.

Commit:
Update on "Build a __torch_dispatch__ class that records torch operator names"

- Created a SchemaCheckTensor torch.tensor subclass to verify that ops match corresponding schema. Right now this keeps a running list of ops which can be cleared and displayed with start_recording() and display_ops() respectively within the schema_check_tensor class
- Created a test_schema_check test suite for testing SchemaCheckTensor and related functionality
- Specifically tested that SchemaCheckTensor acts/operates like a normal tensor and that SchemaCheckTensor writes ops in expected order. 

[ghstack-poisoned]

goldenxuett commented:
@pytorchbot merge
